lib/options.go

```diff
@@ -44,6 +47,238 @@ const DefaultScenarioName = "default"
+func (d *NullDNSPolicy) UnmarshalJSON(data []byte) error {
+	if bytes.Equal(data, []byte(`null`)) {
+		return nil
+	}
+	if err := json.Unmarshal(data, &d.DNSPolicy); err != nil {
+		return err
+	}
+	d.Valid = true
+	return nil
+}
+
+// MarshalJSON returns the JSON representation of d
+func (d NullDNSPolicy) MarshalJSON() ([]byte, error) {
+	if !d.Valid {
+		return []byte(`null`), nil
+	}
+	return json.Marshal(d.DNSPolicy)
+}
+
+// DNSSelect is the strategy to use when picking a single IP if more than one
+// is returned for a host name.
+//go:generate enumer -type=DNSSelect -transform=kebab -trimprefix DNS -output dns_select_gen.go
+type DNSSelect uint8
+
+const (
+	// DNSFirst returns the first IP from the response.
+	DNSFirst DNSSelect = iota + 1
+	// DNSRoundRobin rotates the IP returned on each lookup.
+	DNSRoundRobin
+	// DNSRandom returns a random IP from the response.
+	DNSRandom
+)
+
+// UnmarshalJSON converts JSON data to a valid DNSSelect
+func (d *DNSSelect) UnmarshalJSON(data []byte) error {
+	if bytes.Equal(data, []byte(`null`)) {
+		return nil
+	}
+	var s string
+	if err := json.Unmarshal(data, &s); err != nil {
+		return err
+	}
+	v, err := DNSSelectString(s)
+	if err != nil {
+		return err
+	}
+	*d = v
+	return nil
+}
+
+// MarshalJSON returns the JSON representation of d
+func (d DNSSelect) MarshalJSON() ([]byte, error) {
+	return json.Marshal(d.String())
+}
+
+// NullDNSSelect is a nullable wrapper around DNSSelect, required for the
+// current configuration system.
+type NullDNSSelect struct {
+	DNSSelect
+	Valid bool
+}
+
+// UnmarshalJSON converts JSON data to a valid NullDNSStratey
+func (d *NullDNSSelect) UnmarshalJSON(data []byte) error {
+	if bytes.Equal(data, []byte(`null`)) {
+		return nil
+	}
+	if err := json.Unmarshal(data, &d.DNSSelect); err != nil {
+		return err
+	}
+	d.Valid = true
+	return nil
+}
+
+// MarshalJSON returns the JSON representation of d
+func (d NullDNSSelect) MarshalJSON() ([]byte, error) {
+	if !d.Valid {
+		return []byte(`null`), nil
+	}
+	return json.Marshal(d.DNSSelect)
+}
+
+// DefaultDNSConfig returns the default DNS configuration.
+func DefaultDNSConfig() DNSConfig {
+	return DNSConfig{
+		TTL:    null.NewString("5m", false),
+		Select: NullDNSSelect{DNSRandom, false},
+		Policy: NullDNSPolicy{DNSpreferIPv4, false},
+	}
+}
+
+// String implements fmt.Stringer.
+func (c DNSConfig) String() string {
+	out := make([]string, 0, 2)
+	out = append(out, fmt.Sprintf("ttl=%s", c.TTL.String))
+	out = append(out, fmt.Sprintf("select=%s", c.Select.String()))
+	out = append(out, fmt.Sprintf("policy=%s", c.Policy.String()))
+	return strings.Join(out, ",")
+}
+
+// MarshalJSON implements json.Marshaler.
+func (c DNSConfig) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		TTL    null.String   `json:"ttl"`
+		Select NullDNSSelect `json:"select"`
+		Policy NullDNSPolicy `json:"policy"`
+	}{
+		TTL:    c.TTL,
+		Select: c.Select,
+		Policy: c.Policy,
+	})
+}
```

Comment:
Why ... just not setting `json:"-"` on Valid ? 
Also ... hm ... I do think you should be using NullDNSConfig so that you don't actually save the default settings in `metadata.json` ?

Reply:
It hadn't occurred to me 😄 I'll give it a try, thanks.

Reply:
Resolved by 54290d8c.

Though not sure what you mean with `NullDNSConfig`, as the current `DNSConfig` is essentially that except for the name :) So the default settings aren't saved in `metadata.json` if they're not specified:

```
> go run main.go archive 'github.com/loadimpact/k6/samples/http_get.js' && (tar xfO archive.tar metadata.json | jq '.options.dns')
{
  "ttl": null,
  "select": null,
  "policy": null
}
```